services/motion/builtin/builtin.go

```diff
@@ -122,18 +127,31 @@ func (ms *builtIn) Reconfigure(
 	ms.movementSensors = movementSensors
 	ms.slamServices = slamServices
 	ms.components = components
+	if ms.state != nil {
+		ms.state.Stop()
+	}
+	ms.state = state.NewState(ms.logger)
 	return nil
 }
 
 type builtIn struct {
 	resource.Named
-	resource.TriviallyCloseable
 	fsService       framesystem.Service
+	closed          atomic.Bool
+	mu              sync.RWMutex
```

Comment:
I moved the mutex above everything it protects, following the mutext convention.

Reply:
I changed the mutext to a RWMutex as that reads don't need to block each other & changed its name to `mu` (short for mutex) to conform to the convention elsewhere in the go ecosystem & RDK.

Reply:
yes, moved